SU2_CFD/src/python_wrapper_structure.cpp

```diff
@@ -908,6 +909,49 @@ void CFluidDriver::SetInitialMesh() {
   //}
 }
 
+void CSinglezoneDriver::SetInitialMesh() {
+
+  DynamicMeshUpdate(0);
+
+  SU2_OMP_PARALLEL {
+    // Overwrite fictious velocities
+    SU2_OMP_FOR_STAT(omp_chunk_size)
+    for (unsigned long iPoint = 0; iPoint < geometry_container[ZONE_0][INST_0][MESH_0]->GetnPointDomain(); iPoint++) {
+      for (unsigned short iDim = 0; iDim < geometry_container[ZONE_0][INST_0][MESH_0]->GetnDim(); iDim++) {
+        su2double GridVel = 0.0;
+        geometry_container[ZONE_0][INST_0][MESH_0]->nodes->SetGridVel(iPoint, iDim, GridVel);
+      }
+    }
+
+    geometry_container[ZONE_0][INST_0][MESH_0]->InitiateComms(geometry_container[ZONE_0][INST_0][MESH_0], config_container[ZONE_0], GRID_VELOCITY);
+    geometry_container[ZONE_0][INST_0][MESH_0]->CompleteComms(geometry_container[ZONE_0][INST_0][MESH_0], config_container[ZONE_0], GRID_VELOCITY);
+
+    for (iMesh = 1u; iMesh <= config_container[ZONE_0]->GetnMGLevels(); iMesh++) {
+      SU2_OMP_FOR_STAT(roundUpDiv(geometry_container[ZONE_0][INST_0][iMesh]->GetnPoint(),omp_get_max_threads()))
+      for (unsigned long Point_Coarse = 0; Point_Coarse < geometry_container[ZONE_0][INST_0][iMesh]->GetnPoint(); Point_Coarse++) {
+
+        /*--- Overwrite fictitious velocities ---*/
+        su2double Grid_Vel[3] = {0.0, 0.0, 0.0};
+
+        /*--- Set the grid velocity for this coarse node. ---*/
+        for (unsigned short iDim = 0; iDim < geometry_container[ZONE_0][INST_0][MESH_0]->GetnDim(); iDim++)
+          geometry_container[ZONE_0][INST_0][iMesh]->nodes->SetGridVel(Point_Coarse, iDim, Grid_Vel[iDim]);
+      }
+    }
```

Comment:
Here you can run the loop over grids starting from 0 which would cover the MESH_0 case.
And since the loop is over nPoint (instead of nPointDomain) the communications are not required.

Reply:
You can use the version of SetGridVel that takes an array, so you dont need the loop over nDim